homeassistant/components/tts/__init__.py

```diff
@@ -446,18 +469,18 @@ def load_speech():
         self._async_store_to_memcache(key, filename, data)
 
     @callback
-    def _async_store_to_memcache(self, key, filename, data):
+    def _async_store_to_memcache(self, key: str, filename: str, data: bytes) -> None:
         """Store data to memcache and set timer to remove it."""
         self.mem_cache[key] = {MEM_CACHE_FILENAME: filename, MEM_CACHE_VOICE: data}
 
         @callback
-        def async_remove_from_mem():
+        def async_remove_from_mem() -> None:
             """Cleanup memcache."""
             self.mem_cache.pop(key, None)
 
         self.hass.loop.call_later(self.time_memory, async_remove_from_mem)
 
-    async def async_read_tts(self, filename):
+    async def async_read_tts(self, filename: str) -> TtsAudioType:
```

Comment:
I'm not sure we can use TtsAudioType as return value typing since the data doesn't seem to be optional.

Reply:
I've adjusted this to return a `tuple[str | None, bytes]`